### DISCUSSION TITLE
【LINE】自分で作ったBOTを、他の人がテストしたい。

### DISCUSSION THREAD

**Discussion by SuzumitsuTs1103** [1ft-seabass/ma-hackathon-qa-and-knowledge-202210#28] [Q&A]
@1ft-seabass 
LINE botを作りました。他の人とそのbotはお友達になったのですが、
リッチメニューのイベントを拾うことができません。

他のメンバにテストを実施させる設定は可能ですか？

⇨Pythonで作っています。
⇨参考例でもOKです

**Comment by 1ft-seabass** [comment]
@SuzumitsuTs1103 リッチメニューですが、こちらってリッチメニューボタンを押したイベントですかねー？

**Reply by 1ft-seabass** [reply]
あ、あとできたら Slack にも投稿してほしいです～。駆け込み寺ですかね。

**Reply by SuzumitsuTs1103** [reply]
リッチメニュでテキストを発火して、それをトリガにしています。
他の人が操作をすると、テキストまでは発火しますが、処理が動いていないですね。

Slackの件、承知しました

**Reply by 1ft-seabass** [reply]
@SuzumitsuTs1103 回答たすかります！なるほどテキストは届くってことですね。

それでしたら、LINE BOT 側でテキストメッセージとして届いているはずなので、それを捉えるとどうでしょう？

**Reply by 1ft-seabass** [reply]
参考文献

リッチメニューオブジェクト
https://developers.line.biz/ja/reference/messaging-api/#rich-menu-response-object

リッチメニューアクション
https://developers.line.biz/ja/reference/messaging-api/#action-objects

**Reply by 1ft-seabass** [reply]
@SuzumitsuTs1103 ↑

**Reply by 1ft-seabass** [reply]
@SuzumitsuTs1103 もし、ユーザーごとに誰が押したかを判定した場合は、ボタンのアクションで作成されたメッセージオブジェクトに入っているユーザー ID で判別できるはずですー。
